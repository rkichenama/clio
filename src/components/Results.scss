.results {
  overflow-y: auto;
  counter-reset: result;

  .result {
    counter-increment: result;
    margin-left: 2em;


    &:nth-of-type(even) {
      background-color: rgba(0, 0, 0, 0.2);
    }

    > div {
      position: relative;
      &:nth-child(1) {
        font-weight: bold;

        &::before {
          content: counter(result) ".";
          position: absolute;
          display: inline-block;
          left: -2em;
        }
      }
    }

    > section {
      display: grid;
      grid-template-columns: 48px 1fr;

      > button {
        height: 48px;
      }
    }

    .close,
    .open {
      padding: 0 3em;
    }
    .close {
      padding-right: 0;
    }
  }
}