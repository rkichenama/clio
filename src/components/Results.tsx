import React from 'react';
import { SearchContext } from '../Context';
import firstNChars from '../shared/domparse';

import './Results.scss';

type ResultType = Pick<Job, 'company' | 'name' | 'locations' | 'contents'>;
const Result: React.FC<ResultType> = ({
  company, contents, locations, name, refs: { landingPage }
}) => {
  const [ expanded, toggleExpanded ] = React.useState(false);

  return (
    <article className='result'>
      <div>
        <a href={landingPage} target='_blank' rel='noopener'>{ name }</a>
      </div>
      <div>{ company.name }</div>
      <div className='flexible-container'>{ locations.map(({ name }, idx) => (
        <span key={`${idx}-${name}`}>{ name }</span>
      )) }</div>
      <section>
        <button onClick={() => toggleExpanded(open => !open)}>
          { expanded ? 'less' : 'more' }
        </button>
        <div className={ expanded ? 'open' : 'closed' } dangerouslySetInnerHTML={{__html: expanded ? contents : firstNChars(contents) }} />
      </section>
    </article>
  );
}

const Results: React.FC<any> = () => {
  const { loading, loaded, data, error } = React.useContext(SearchContext);

  return (
    <section className='results'>
      { error
        ? ('An error occurred')
        : loading && !loaded
          ? ('Loading...')
          : !loading && !loaded
            ? ('please try doing a search')
            : data.map(job => (
              <Result key={job.id} {...job} />
            ))
      }
    </section>
  );
};

export default Results;
