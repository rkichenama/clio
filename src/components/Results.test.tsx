import React from 'react';
import { shallow, mount } from 'enzyme';
import Results, { Result } from './Results';
import { createMock } from 'ts-auto-mock';

jest.mock('../shared/domparse.ts')

describe('Result', () => {
  let job: Job;

  beforeEach(() => {
    job = createMock<Job>();
  });

  it('should render', () => {
    expect(shallow(<Result {...job} />)).toMatchSnapshot();
  });
});

describe('Results', () => {
  let job: Job;
  let useContextSpy;

  beforeAll(() => {
    useContextSpy = jest.spyOn(React, 'useContext')
      .mockImplementation(() => ({
        loading: false, loaded: true, data: []
      }));
  })
  beforeEach(() => {
    job = createMock<Job>();
    useContextSpy.mockClear();
  });

  it('should render', () => {
    useContextSpy.mockImplementationOnce(() => ({
      loading: false, loaded: true, data: [ job ]
    }));
    expect(shallow(<Results />)).toMatchSnapshot();
  });
});
