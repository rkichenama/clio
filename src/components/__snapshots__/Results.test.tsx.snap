// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render 1`] = `
<article
  className="result"
>
  <div>
    <a
      href=""
      rel="noopener"
      target="_blank"
    />
  </div>
  <div />
  <div
    className="flexible-container"
  />
  <section>
    <button
      onClick={[Function]}
    >
      more
    </button>
    <div
      className="closed"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </section>
</article>
`;

exports[`Results should render 1`] = `
<section
  className="results"
>
  <Result
    categories={Array []}
    company={
      Object {
        "id": 0,
        "name": "",
        "shortName": "",
      }
    }
    contents=""
    id={0}
    key="0"
    levels={Array []}
    locations={Array []}
    name=""
    publicationDate=""
    refs={
      Object {
        "landingPage": "",
      }
    }
    shortName=""
    tags={Array []}
    type="internal"
  />
</section>
`;
